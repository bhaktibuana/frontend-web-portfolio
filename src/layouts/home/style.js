import styled from "styled-components";

export const Section = styled.section`
  width: 100%;
  height: auto;
  min-height: 800px;
  background-color: ${({ color }) => color.bgLight};
  display: flex;
  justify-content: center;
  align-items: center;
`;
