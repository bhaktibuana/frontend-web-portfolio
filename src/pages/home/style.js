import styled from "styled-components";

export const Container = styled.div`
  width: 100%;
  height: auto;
  min-height: 100vh;
`;

export const Section = styled.section`
  width: 100%;
  height: 800px;
  background-color: ${({ color }) => color};
`;
