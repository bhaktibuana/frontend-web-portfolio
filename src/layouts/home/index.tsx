import React from "react";
import { colorPalette } from "../../constants/colorPalette";
import { Section } from "./style";

interface ComponentProps {
  id: string;
}

const HomeLayout = ({ id }: ComponentProps) => {
  return (
    <>
      <Section id={id} color={colorPalette}>
        <h1 style={{ color: "gray", fontSize: "4rem" }}>COMING SOON</h1>
      </Section>
    </>
  );
};

export default HomeLayout;
