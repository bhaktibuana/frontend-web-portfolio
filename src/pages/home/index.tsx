import React, { useEffect } from "react";
import { animateScroll } from "react-scroll";
import {
  MdHome,
  MdAssignmentInd,
  MdWorkHistory,
  MdOutlineAppRegistration,
  MdArticle,
  MdMail,
} from "react-icons/md";
import { BsFillFileEarmarkCodeFill } from "react-icons/bs";
import HomeNavbarLayout from "../../layouts/navigation/homeNavbar";
import HomeLayout from "../../layouts/home";
import HomeSidebarLayout from "../../layouts/navigation/homeSidebar";
import { Container, Section } from "./style";

export const navButtonArr = [
  {
    text: "Home",
    scrollTo: "home",
    icon: <MdHome size={24} />,
  },
  {
    text: "About",
    scrollTo: "about",
    icon: <MdAssignmentInd size={24} />,
  },
  {
    text: "Skills",
    scrollTo: "skills",
    icon: <BsFillFileEarmarkCodeFill size={24} />,
  },
  {
    text: "Experience",
    scrollTo: "experience",
    icon: <MdWorkHistory size={24} />,
  },
  {
    text: "Portfolio",
    scrollTo: "protfolio",
    icon: <MdOutlineAppRegistration size={24} />,
  },
  {
    text: "Blog",
    scrollTo: "blog",
    icon: <MdArticle size={24} />,
  },
  {
    text: "Contact",
    scrollTo: "contact",
    icon: <MdMail size={24} />,
  },
];

const HomePage = () => {
  const scrollToTop = () => {
    animateScroll.scrollToTop();
  };

  useEffect(() => {
    scrollToTop();
  }, []);

  return (
    <>
      <Container data-testid="homePage">
        <HomeSidebarLayout navButton={navButtonArr} />
        <HomeNavbarLayout navButton={navButtonArr} />

        <HomeLayout id="home" />
        <Section id="about" color="green" />
        <Section id="skills" color="blue" />
      </Container>
    </>
  );
};

export default HomePage;
