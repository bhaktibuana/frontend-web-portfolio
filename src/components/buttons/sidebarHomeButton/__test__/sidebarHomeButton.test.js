import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import SidebarHomeButton from "..";
import { navButtonArr } from "../../../../pages/home";

afterEach(() => {
  cleanup();
});

test("should render button", () => {
  render(
    <SidebarHomeButton
      text={navButtonArr[0].text}
      scrollTo={navButtonArr[0].scrollTo}
      icon={navButtonArr[0].icon}
    />
  );
  const buttonElement = screen.getByTestId("sidebarHomeButton");

  expect(buttonElement).toBeInTheDocument();
  expect(buttonElement).toHaveTextContent("Home");
});

test("matches snapshot", () => {
  const tree = renderer
    .create(
      <SidebarHomeButton
        text={navButtonArr[0].text}
        scrollTo={navButtonArr[0].scrollTo}
        icon={navButtonArr[0].icon}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
