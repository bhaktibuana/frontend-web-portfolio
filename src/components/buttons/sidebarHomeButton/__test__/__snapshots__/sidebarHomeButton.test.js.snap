// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<a
  className="sc-beqWaB ipwSuT"
  color={
    Object {
      "bgDark": "#202020",
      "bgLight": "#ffffff",
      "primary": "#14b8a6",
      "secondary": "#1e293b",
      "tertiary": "#f1f5f9",
      "textDark": "#ffffff",
      "textLight": "#202020",
    }
  }
  data-testid="sidebarHomeButton"
  onClick={[Function]}
  style={Object {}}
>
  <div
    className="sc-gueYoa jGxKvQ"
  >
    <svg
      fill="currentColor"
      height={24}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
      />
    </svg>
  </div>
  <p>
    Home
  </p>
</a>
`;
